frontend/src/components/admin/question-creation/PreviewQuestionModal.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useContext } from "react";
+import { Button } from "@chakra-ui/react";
+import { v4 as uuidv4 } from "uuid";
+
+import { EditOutlineIcon } from "../../../assets/icons";
+import QuestionEditorContext from "../../../contexts/QuestionEditorContext";
+import AssessmentExperience from "../../student/AssessmentExperience";
+
+const PreviewQuestionModal = (): React.ReactElement => {
```

Comment:
This is called a `Modal`; does it actually act like one? If this is the page we discussed, maybe `PreviewQuestionPage` instead?

Reply:
Hmm. I'm torn here.

I don't love the idea of calling it a page, since in my mind each of our pages correspond to a unique route, whereas this one doesn't. Additionally, In my opinion, it acts more like a modal since we control its display with a `showModal` state  variable that we toggle. 

However, I also see why it may not be a `Modal` in the traditional sense. Any other suggestions?

Reply:
Any reason why it doesn't get its own route? Seems like we could do some sort of /assessment/<id>/edit -> /assessment/<id>/preview (probably in a future ticket though?)

Reply:
Hmm, it's a less trivial change than it seems, since we don't have unique routes for the `assessment-editor` (i.e the assessment ID is not contained within the URL for the assessment editor).

Because of this, we don't have routing at all for the`question-editor`, (because if you were to go directly to `/assessment-editor/question-editor/`, what would you see? it wouldn't be attached to any assessment). So because of this problem, I implemented the sub-feature of **Preview Question** in the same fashion.

Reply:
Yeah, I'm not saying we should do the path refactor in this pr. In fact, I have a backlog ticket I want to do once we hit our deadlines (we _will_ 🤞) to add this kind of routing to more pages. But I think calling it a "page" component would make sense here.